diff --git a/search_api_db.install b/search_api_db.install
index ae604e3..7948992 100644
--- a/search_api_db.install
+++ b/search_api_db.install
@@ -115,6 +115,9 @@ function search_api_db_update_7103() {
  * Use a single full text table per index.
  */
 function search_api_db_update_7104() {
+  // See issue #2428693
+  module_implements('', FALSE, TRUE);
+
   $servers_query = db_select('search_api_server', 's')
     ->condition('s.class', 'search_api_db_service');
   $servers_query->innerJoin('search_api_index', 'i', 'i.server = s.machine_name');
