diff --git a/views/ip_geoloc_plugin_style_leaflet.inc b/views/ip_geoloc_plugin_style_leaflet.inc
index 54039717a10fee220cbcb41cc2d100e435fb5790..fb19e6f5aad77f9ca422a8d0da3690b41528b699 100644
--- a/views/ip_geoloc_plugin_style_leaflet.inc
+++ b/views/ip_geoloc_plugin_style_leaflet.inc
@@ -108,6 +108,7 @@ class ip_geoloc_plugin_style_leaflet extends views_plugin_style {
     $options['map_options'] = array(
       'contains' => array(
         'maxzoom' => array('default' => 18),
+        'minzoom' => array('default' => 12),
         'zoom' => array('default' => 2),
         'center_lat' => array('default' => ''),
         'center_lon' => array('default' => ''),
@@ -729,6 +730,7 @@ class ip_geoloc_plugin_style_leaflet extends views_plugin_style {
     $module_path = drupal_get_path('module', 'ip_geoloc');
     $marker_path = file_create_url(ip_geoloc_marker_directory());
     $max_zoom = (int) $this->options['map_options']['maxzoom'];
+    $min_zoom = (int) $this->options['map_options']['minzoom'];
     $zoom = max(1, (int)$this->options['map_options']['zoom']);
     $scroll_wheel_zoom = (bool) $this->options['map_options']['scrollwheelzoom'];
     $dragging = (bool) $this->options['map_options']['dragging'];
@@ -937,6 +939,8 @@ class ip_geoloc_plugin_style_leaflet extends views_plugin_style {
       }
     }
     $map['settings']['maxZoom'] = $max_zoom;
+    $map['settings']['minZoom'] = $min_zoom;
+    $map['settings']['maxBounds'] = array([[52.270517, 12.796015],[52.721994, 13.812250]]);
     $map['settings']['scrollWheelZoom'] = $scroll_wheel_zoom;
     $map['settings']['dragging'] = $dragging;

